lib/nativeTimerifyWrap.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
```

Comment:
I have an alternative implementation [here](https://github.com/immobiliare/fastify-metrics/blob/feat_timing_utils_2/lib/nativeTimerifyWrap.js) that uses only one perf observer. I have seen that each observer created is listening to all functions, hence each one will have its call-back triggered event for functions you might not want to measure. So at this pint, with some custom logic I decided to try to have on handler for all of the entries. wdyt?

Reply:
It would be nice to couple it with the finalilzationRegistry to perform cleanup only when the function is garbage collected, but I am not sure it follows the guidelines exposed in Node docs regarding the lifetime of a perf observer.

Reply:
Anyway, these are all things that are not blocking reagardéng the documentation update, benchmarks and types update. At least for now in this PR.

Reply:
As we discussed: I like the approach, let's just check `entry.name` because I'm not sure that will work with arrow function, just in case we have to write this somehow in the docs! gj!

Reply:
Seeing MDN, FinalizationRegistry is supported on Node 16. We could try it. I would say after we add the benchmarks.